File: eCinema/eCinema.Model/Termin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Model
{
    public class Termin
    {
        public int TerminId { get; set; }
        public bool Premijera { get; set; }
        public bool Predpremijera { get; set; }
        public int CijenaKarte { get; set; }
        public DateTime DatumOdrzavanja { get; set; }
        public string VrijemeOdrzavanja { get; set; }

        public int? DvoranaId { get; set; }
        public virtual Dvorana? Dvorana { get; set; }
        public int? FilmId { get; set; }
        public virtual Film? Film { get; set; }
        
    }
}

File: ecinema_desktop/lib/models/termin.dart
import 'package:ecinema_desktop/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
  int terminId;
  bool premijera;
  bool predpremijera;
  int cijenaKarte;
  DateTime datumOdrzavanja;
  String vrijemeOdrzavanja;
  int? dvoranaId;
  Dvorana? dvorana;
  int? filmId;
  Film? film;

  Termin(
      {required this.terminId,
      required this.premijera,
      required this.predpremijera,
      required this.cijenaKarte,
      required this.datumOdrzavanja,
      required this.vrijemeOdrzavanja,
      this.dvoranaId,
      this.dvorana,
      this.filmId,
      this.film});

  factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);
  Map<String, dynamic> toJson() => _$TerminToJson(this);
}
